### PR TITLE
Update Dialog Code

### PR DESCRIPTION
**Pull request by sampoearas**
- Updated code to create/access dialog boxes, post-SP7
- Removed Extra Class Picker Form
- Updated readme with install/usage instructions, instead of separate PDF
- Removed PDF

### CODE REVIEW

File: Import/Razorleaf Class Picker/Import/Method/RL - Select Class On New.xml [hunk 1]
```diff
@@ -59,49 +60,54 @@ if (itemTypeName === '')
...
+    else
+    {
+      //Update the item 
+      theItem.setProperty("classification", res);
+      //Set a flag in the item to denote that a class was selected from this method
+      theItem.setProperty("razorleaf_select_class", "1");
+      
+      //Refresh Form
+      //if main item, refresh; else refresh BOM item
+      if(theItem.getParentItem() === null){
```
Comment (EliJDonahue) [inline Import/Razorleaf Class Picker/Import/Method/RL - Select Class On New.xml]: Why the strange formatting of the if-else?

File: Import/Razorleaf Class Picker/Import/Method/RL - Select Class On New.xml [hunk 2]
```diff
@@ -59,49 +60,54 @@ if (itemTypeName === '')
...
-{
-   //Update the item 
-   this.setProperty("classification", res);
-   //Set a flag in the item to denote that a class was selected from this method
-   this.setProperty("razorleaf_select_class", "1");
-}
-  
+
+// Open the dialog
+var topWnd = top.aras.getMostTopWindowWithAras();
```
Comment (EliJDonahue) [inline Import/Razorleaf Class Picker/Import/Method/RL - Select Class On New.xml]: Just use aras.getMostTopWindowWithAras(). The top.aras object has been deprecated.

**Review by EliJDonahue — CHANGES_REQUESTED** [review]
General notes:
1. Pay attention to formatting and white space in your code. There are some extra tab indentations in some places.
2. Format the Credits section of the readme like our other readmes. You can remove the created on date.

File: Import/Razorleaf Class Picker/Import/Method/RL - Select Class On New.xml (hunk 1 above)
Comment (sampoearas) [reply Import/Razorleaf Class Picker/Import/Method/RL - Select Class On New.xml]: I think that was the method editor being weird about the brackets. Fixed.

File: Import/Razorleaf Class Picker/Import/Method/RL - Select Class On New.xml (hunk 2 above)
Comment (sampoearas) [reply Import/Razorleaf Class Picker/Import/Method/RL - Select Class On New.xml]: I think that was the only top.aras I missed somehow...

**Comment by sampoearas** [comment]
1. I think some of the tab indentations came from trying to align things in the method editor.
2. Credits section updated.

**Comment by sampoearas** [comment]
Issues Resolved.

File: Import/Razorleaf Class Picker/Import/Method/RL - Select Class On New.xml [hunk 3]
```diff
@@ -59,52 +60,59 @@ if (itemTypeName === '')
...
-   //If an item classification is selected from the tree other than the item type, then the leaf class name is return res=Drawing
-
-//Set the selected value on the class structure on the form and apply to the item
-if (res === null || res === '' || res === undefined)
-{
-   //A classification was not selected by the user
-   //Cancel the operation
-   return false;
+function callback(res) {
+	 if (res === null || res === '' || res === undefined)
```
Comment (EliJDonahue) [inline Import/Razorleaf Class Picker/Import/Method/RL - Select Class On New.xml]:
```suggestion
    if (res === null || res === '' || res === undefined) {
```

File: Import/Razorleaf Class Picker/Import/Method/RL - Select Class On New.xml [hunk 4]
```diff
@@ -59,52 +60,59 @@ if (itemTypeName === '')
...
+    {
```
Comment (EliJDonahue) [inline Import/Razorleaf Class Picker/Import/Method/RL - Select Class On New.xml]:
```suggestion
```

File: Import/Razorleaf Class Picker/Import/Method/RL - Select Class On New.xml [hunk 5]
```diff
@@ -59,52 +60,59 @@ if (itemTypeName === '')
...
+      //A classification was not selected by the user
+      //Cancel the operation
+      return false;
+    }
```
Comment (EliJDonahue) [inline Import/Razorleaf Class Picker/Import/Method/RL - Select Class On New.xml]:
```suggestion
    } else {
```

File: Import/Razorleaf Class Picker/Import/Method/RL - Select Class On New.xml [hunk 6]
```diff
@@ -59,52 +60,59 @@ if (itemTypeName === '')
...
+    else
```
Comment (EliJDonahue) [inline Import/Razorleaf Class Picker/Import/Method/RL - Select Class On New.xml]:
```suggestion
```

File: Import/Razorleaf Class Picker/Import/Method/RL - Select Class On New.xml [hunk 7]
```diff
@@ -59,52 +60,59 @@ if (itemTypeName === '')
...
+    {
```
Comment (EliJDonahue) [inline Import/Razorleaf Class Picker/Import/Method/RL - Select Class On New.xml]:
```suggestion
```

File: Import/Razorleaf Class Picker/Import/Method/RL - Select Class On New.xml [hunk 8]
```diff
@@ -59,52 +60,59 @@ if (itemTypeName === '')
...
+      //Update the item 
+      theItem.setProperty("classification", res);
+      //Set a flag in the item to denote that a class was selected from this method
+      theItem.setProperty("razorleaf_select_class", "1");
+      
+      //Refresh Form
+      //if main item, refresh; else refresh BOM item
+      if(theItem.getParentItem() === null){
```
Comment (EliJDonahue) [inline Import/Razorleaf Class Picker/Import/Method/RL - Select Class On New.xml]:
```suggestion
      if (theItem.getParentItem() === null) {
```

File: Import/Razorleaf Class Picker/Import/Method/RL - Select Class On New.xml [hunk 9]
```diff
@@ -59,52 +60,59 @@ if (itemTypeName === '')
...
+      aras.uiReShowItem(theItem.getID(), theItem.getID());}
```
Comment (EliJDonahue) [inline Import/Razorleaf Class Picker/Import/Method/RL - Select Class On New.xml]:
```suggestion
      aras.uiReShowItem(theItem.getID(), theItem.getID());
```

File: Import/Razorleaf Class Picker/Import/Method/RL - Select Class On New.xml [hunk 10]
```diff
@@ -59,52 +60,59 @@ if (itemTypeName === '')
...
+      else {
```
Comment (EliJDonahue) [inline Import/Razorleaf Class Picker/Import/Method/RL - Select Class On New.xml]:
```suggestion
      } else {
```

File: Import/Razorleaf Class Picker/Import/Method/RL - Select Class On New.xml [hunk 11]
```diff
@@ -59,52 +60,59 @@ if (itemTypeName === '')
...
+	aras.uiReShowItem(theItem.getParentItem().getID(), theItem.getParentItem().getID());
```
Comment (EliJDonahue) [inline Import/Razorleaf Class Picker/Import/Method/RL - Select Class On New.xml]:
```suggestion
	    aras.uiReShowItem(theItem.getParentItem().getID(), theItem.getParentItem().getID());
```

**Review by EliJDonahue — CHANGES_REQUESTED** [review]

**Comment by sampoearas** [comment]
Applied white space suggestions

**Review by EliJDonahue — APPROVED** [review]

**Pull request merged**